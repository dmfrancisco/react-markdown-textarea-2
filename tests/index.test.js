import React from 'react';
import { shallow } from 'enzyme';
import MarkdownIt from 'markdown-it';

import MarkdownTextarea from '../src';

const markdown = new MarkdownIt();

it('renders correctly with only required props', () => {
  const element = shallow(
    <MarkdownTextarea
      render={value => markdown.render(value)}
    />
  );
  expect(element.getNodes()).toMatchSnapshot();
});

it('displays toolbar on focus', () => {
  const element = shallow(
    <MarkdownTextarea
      render={value => markdown.render(value)}
    />
  );
  element.find('.MarkdownTextarea-write').simulate('focus');
  expect(element.getNodes()).toMatchSnapshot();
});

it('supports preview', () => {
  const element = shallow(
    <MarkdownTextarea
      value="Hello _World_"
      render={value => markdown.render(value)}
    />
  );
  element.find('.MarkdownTextarea-write').simulate('focus');
  element.find('.MarkdownTextarea-action').at(1).simulate('click');
  expect(element.getNodes()).toMatchSnapshot();
});
