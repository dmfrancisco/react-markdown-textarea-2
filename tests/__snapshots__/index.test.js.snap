// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`displays toolbar on focus 1`] = `
Array [
  <div
    className="MarkdownTextarea is-focused"
>
    <div
        className="MarkdownTextarea-toolbar"
    >
        <button
            className="MarkdownTextarea-action is-active"
            onClick={[Function]}
        >
            Write
        </button>
        <button
            className="MarkdownTextarea-action "
            onClick={[Function]}
        >
            Preview
        </button>
    </div>
    <TextareaAutosize
        className="MarkdownTextarea-write"
        inputRef={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHeightChange={[Function]}
        style={
            Object {
                "display": "block",
              }
        }
        useCacheForDOMMeasurements={false}
        value=""
    />
</div>,
]
`;

exports[`renders correctly with only required props 1`] = `
Array [
  <div
    className="MarkdownTextarea "
>
    <TextareaAutosize
        className="MarkdownTextarea-write"
        inputRef={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHeightChange={[Function]}
        style={
            Object {
                "display": "block",
              }
        }
        useCacheForDOMMeasurements={false}
        value=""
    />
</div>,
]
`;

exports[`supports preview 1`] = `
Array [
  <div
    className="MarkdownTextarea is-focused"
>
    <div
        className="MarkdownTextarea-toolbar"
    >
        <button
            className="MarkdownTextarea-action "
            onClick={[Function]}
        >
            Write
        </button>
        <button
            className="MarkdownTextarea-action is-active"
            onClick={[Function]}
        >
            Preview
        </button>
    </div>
    <TextareaAutosize
        className="MarkdownTextarea-write"
        inputRef={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onHeightChange={[Function]}
        style={
            Object {
                "display": "none",
              }
        }
        useCacheForDOMMeasurements={false}
        value="Hello _World_"
    />
    <div
        className="MarkdownTextarea-preview"
        dangerouslySetInnerHTML={
            Object {
                "__html": "<p>Hello <em>World</em></p>
            ",
              }
        }
        style={
            Object {
                "minHeight": null,
              }
        }
    />
</div>,
]
`;
